assets/javascripts/discourse/initializers/set-category-drop-options.js

```diff
@@ -11,23 +11,25 @@ export default {
     const siteSettings = container.lookup("service:site-settings");
     if (siteSettings.discourse_global_filter_enabled) {
       withPluginApi("1.3.0", (api) => {
-        api.modifyClass("controller:discovery/categories", {
+        const modification = {
           pluginId: PLUGIN_ID,
 
           init() {
             this._super(...arguments);
             setCategoryDropOptionsPerGlobalFilter(api);
           },
-        });
+        };
 
-        api.modifyClass("controller:tag-show", {
-          pluginId: PLUGIN_ID,
+        api.modifyClass("controller:discovery/categories", modification);
 
-          init() {
-            this._super(...arguments);
-            setCategoryDropOptionsPerGlobalFilter(api);
-          },
-        });
+        // TODO: Remove once https://github.com/discourse/discourse/pull/22622 is merged
+        api.modifyClass("controller:tag-show", modification);
+
+        const owner = api.container.owner;
+        if (owner.resolveRegistration("controller:discovery/list")) {
+          // TODO: remove conditional once https://github.com/discourse/discourse/pull/22622 is merged
+          api.modifyClass("controller:discovery/list", modification);
+        }
```

Comment:
Do you need to check for the registration? To avoid warning messages you could use the `ignoreMissing` option:

```js
api.modifyClass("controller:discovery/list", modification, { ignoreMissing: true });
```

Reply:
That's much cleaner - thanks!